pkg/jobs/registry.go

```diff
@@ -275,6 +278,11 @@ func (r *Registry) MetricsStruct() *Metrics {
 	return &r.metrics
 }
 
+// MetricsRegistry returns the metrics registry that stores job-related metrics.
+func (r *Registry) MetricsRegistry() *metric.Registry {
```

Comment:
I'd have a slight preference for putting this directly in JobExecContext or in ExecutorConfig since those is supposed the thing jobs/sql code need to execute, whereas jobs.Registry is supposed to just be the thing that executes _them_ rather than something they depend on. Obviously it already is used as a dep in a couple places so this is already a little circular and I not something I feel strongly about, but this feels like it'd be more at home in ExecutorConfig than hanging off of job registry.